src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -853,6 +855,25 @@ public ResponseEntity<List<User>> getAllInstructorsInCourse(@PathVariable Long c
         return courseService.getAllUsersInGroup(course, course.getInstructorGroupName());
     }
 
+    /**
+     * GET /courses/:courseId/search-users : search users for a given course within all groups.
+     *
+     * @param courseId    the id of the course for which to search users
+     * @param nameOfUser  the name by which to search users
+     * @return the ResponseEntity with status 200 (OK) and with body all users
+     */
+    @GetMapping("/courses/{courseId}/search-other-users")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<List<User>> searchOtherUsersInCourse(@PathVariable long courseId, @RequestParam("nameOfUser") String nameOfUser) {
+        // restrict result size by only allowing reasonable searches
+        if (nameOfUser.length() < 3) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Query param 'name' must be three characters or longer.");
+        }
+        Course course = courseRepository.findByIdElseThrow(courseId);
+
```

Comment:
According to the comment in CourseService, I think it would be better to check here if the user has at least students' permission for this course.